sql/catalog/lease: TestLeaseRenewedPeriodically failed

sql/catalog/lease.TestLeaseRenewedPeriodically [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3397818&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3397818&tab=artifacts#/) on master @ [63679e7bf4a0244c6e46381751543cfe3f32388c](https://github.com/cockroachdb/cockroach/commits/63679e7bf4a0244c6e46381751543cfe3f32388c):


```
I210902 06:51:45.860134 38313 util/log/event_log.go:32  [n1,intExec=set-setting] 1726 ={"Timestamp":1630565505858024182,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-set-setting","PlaceholderValues":["'21.1-160'"],"SettingName":"version","Value":"21.1-160"}
I210902 06:51:45.871942 38556 util/log/event_log.go:32  [n1,intExec=initializeClusterSecret] 1727 ={"Timestamp":1630565505869449872,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"cluster.secret\" = gen_random_uuid()::STRING","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-initializeClusterSecret","SettingName":"cluster.secret","Value":"8936c149-2907-49dd-a7d4-9fdaba6c1f1a"}
I210902 06:51:45.874981 38770 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 1728 ={"Timestamp":1630565505873127942,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS defaultdb","Tag":"CREATE DATABASE","User":"root","DescriptorID":50,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"defaultdb"}
I210902 06:51:45.874962 38251 sql/catalog/lease/lease.go:906  [n1] 1729  skipping update of database:<name:"defaultdb" id:50 modification_time:<wall_time:1630565505873127942 > version:1 privileges:<users:<user_proto:"admin" privileges:2 > users:<user_proto:"root" privileges:2 > owner_proto:"root" version:2 > state:PUBLIC offline_reason:"" default_privileges:<> > due to knob: ignore gossip update
I210902 06:51:45.875601 37997 kv/kvserver/queue.go:581  [n1,s1,r35/1:/Table/{39-40}] 1730  rate limited in MaybeAdd (merge): throttled on async limiting semaphore
I210902 06:51:45.876207 38143 sql/sqlliveness/slstorage/slstorage.go:393  [n1] 1731  inserted sqlliveness session 6f38f4dc239340da91334631b20acf99
I210902 06:51:45.876255 38143 sql/sqlliveness/slinstance/slinstance.go:180  [n1] 1732  created new SQL liveness session 6f38f4dc239340da91334631b20acf99
I210902 06:51:45.876894 37985 3@vendor/github.com/cockroachdb/pebble/event.go:617  [n1,pebble,s1] 1733  [JOB 3] WAL created 000004
I210902 06:51:45.877538 38787 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 1734 ={"Timestamp":1630565505875172019,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS postgres","Tag":"CREATE DATABASE","User":"root","DescriptorID":51,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"postgres"}
I210902 06:51:45.877885 38256 spanconfig/spanconfigmanager/manager.go:127  [n1] 1735  started auto span config reconciliation job
I210902 06:51:45.879180 38251 sql/catalog/lease/lease.go:906  [n1] 1736  skipping update of database:<name:"postgres" id:51 modification_time:<wall_time:1630565505875172019 > version:1 privileges:<users:<user_proto:"admin" privileges:2 > users:<user_proto:"root" privileges:2 > owner_proto:"root" version:2 > state:PUBLIC offline_reason:"" default_privileges:<> > due to knob: ignore gossip update
I210902 06:51:45.880934 38715 jobs/adopt.go:247  [-] 1737  job 689750362850000897: resuming execution
I210902 06:51:45.883278 38718 jobs/registry.go:1150  [n1] 1738  AUTO SPAN CONFIG RECONCILIATION job 689750362850000897: stepping through state running with error: <nil>
I210902 06:51:45.890845 37599 server/server_sql.go:1087  [n1] 1739  done ensuring all necessary startup migrations have run
I210902 06:51:45.890972 39044 jobs/job_scheduler.go:349  [n1] 1740  waiting 2m0s before scheduled jobs daemon start
I210902 06:51:45.892598 39046 server/auto_upgrade.go:55  [n1] 1741  no need to upgrade, cluster already at the newest version
I210902 06:51:45.899826 37978 3@vendor/github.com/cockroachdb/pebble/event.go:617  [n1,pebble,s1] 1742  [JOB 4] WAL created 000005
I210902 06:51:45.901254 37599 1@server/server.go:2329  [n1] 1743  serving sql connections
I210902 06:51:45.912104 38918 5@util/log/event_log.go:32  [n1,client=127.0.0.1:55714,hostssl,user=root] 1744 ={"Timestamp":1630565505909830782,"EventType":"create_database","Statement":"CREATE DATABASE t","Tag":"CREATE DATABASE","User":"root","DescriptorID":52,"DatabaseName":"t"}
I210902 06:51:45.912066 38251 sql/catalog/lease/lease.go:906  [n1] 1745  skipping update of database:<name:"t" id:52 modification_time:<wall_time:1630565505909830782 > version:1 privileges:<users:<user_proto:"admin" privileges:2 > users:<user_proto:"root" privileges:2 > owner_proto:"root" version:2 > state:PUBLIC offline_reason:"" default_privileges:<> > due to knob: ignore gossip update
I210902 06:51:45.920627 38918 5@util/log/event_log.go:32  [n1,client=127.0.0.1:55714,hostssl,user=root] 1746 ={"Timestamp":1630565505912249749,"EventType":"create_table","Statement":"CREATE TABLE t.public.test1 (k CHAR PRIMARY KEY, v CHAR)","Tag":"CREATE TABLE","User":"root","DescriptorID":53,"TableName":"t.public.test1"}
I210902 06:51:45.920627 38251 sql/catalog/lease/lease.go:906  [n1] 1747  skipping update of table:<name:"test1" id:53 version:1 modification_time:<wall_time:1630565505912249749 > parent_id:52 unexposed_parent_schema_id:29 columns:<name:"k" id:1 type:<InternalType:<family:StringFamily width:1 precision:0 locale:"" visible_type:0 oid:1042 time_precision_is_set:false > TypeMeta:<Version:0 > > nullable:false hidden:false inaccessible:false generated_as_identity_type:NOT_IDENTITY_COLUMN virtual:false pg_attribute_num:0 alter_column_type_in_progress:false system_column_kind:NONE > columns:<name:"v" id:2 type:<InternalType:<family:StringFamily width:1 precision:0 locale:"" visible_type:0 oid:1042 time_precision_is_set:false > TypeMeta:<Version:0 > > nullable:true hidden:false inaccessible:false generated_as_identity_type:NOT_IDENTITY_COLUMN virtual:false pg_attribute_num:0 alter_column_type_in_progress:false system_column_kind:NONE > next_column_id:3 families:<name:"primary" id:0 column_names:"k" column_names:"v" column_ids:1 column_ids:2 default_column_id:2 > next_family_id:1 primary_index:<name:"primary" id:1 unique:true version:4 key_column_names:"k" key_column_directions:ASC store_column_names:"v" key_column_ids:1 store_column_ids:2 foreign_key:<table:0 index:0 name:"" validity:Validated shared_prefix_len:0 on_delete:NO_ACTION on_update:NO_ACTION match:SIMPLE > interleave:<> partitioning:<num_columns:0 num_implicit_columns:0 > type:FORWARD created_explicitly:false encoding_type:1 sharded:<is_sharded:false name:"" shard_buckets:0 > disabled:false geo_config:<> predicate:"" > next_index_id:2 privileges:<users:<user_proto:"admin" privileges:2 > users:<user_proto:"root" privileges:2 > owner_proto:"root" version:2 > next_mutation_id:1 format_version:3 state:PUBLIC offline_reason:"" view_query:"" is_materialized_view:false new_schema_change_job_id:0 drop_time:0 replacement_of:<id:0 time:<> > audit_mode:DISABLED drop_job_id:0 create_query:"" create_as_of_time:<wall_time:1630565505912249749 > temporary:false partition_all_by:false > due to knob: ignore gossip update
I210902 06:51:45.922343 39309 kv/kvserver/replica_command.go:396  [n1,split,s1,r43/1:/{Table/47-Max}] 1748  initiating a split of this range at key /Table/53 [r44] (span config)
I210902 06:51:45.923456 38918 5@util/log/event_log.go:32  [n1,client=127.0.0.1:55714,hostssl,user=root] 1749 ={"Timestamp":1630565505920835476,"EventType":"create_table","Statement":"CREATE TABLE t.public.test2 ()","Tag":"CREATE TABLE","User":"root","DescriptorID":54,"TableName":"t.public.test2"}
I210902 06:51:45.923418 38251 sql/catalog/lease/lease.go:906  [n1] 1750  skipping update of table:<name:"test2" id:54 version:1 modification_time:<wall_time:1630565505920835476 > parent_id:52 unexposed_parent_schema_id:29 columns:<name:"rowid" id:1 type:<InternalType:<family:IntFamily width:64 precision:0 locale:"" visible_type:0 oid:20 time_precision_is_set:false > TypeMeta:<Version:0 > > nullable:false default_expr:"unique_rowid()" hidden:true inaccessible:false generated_as_identity_type:NOT_IDENTITY_COLUMN virtual:false pg_attribute_num:0 alter_column_type_in_progress:false system_column_kind:NONE > next_column_id:2 families:<name:"primary" id:0 column_names:"rowid" column_ids:1 default_column_id:0 > next_family_id:1 primary_index:<name:"primary" id:1 unique:true version:4 key_column_names:"rowid" key_column_directions:ASC key_column_ids:1 foreign_key:<table:0 index:0 name:"" validity:Validated shared_prefix_len:0 on_delete:NO_ACTION on_update:NO_ACTION match:SIMPLE > interleave:<> partitioning:<num_columns:0 num_implicit_columns:0 > type:FORWARD created_explicitly:false encoding_type:1 sharded:<is_sharded:false name:"" shard_buckets:0 > disabled:false geo_config:<> predicate:"" > next_index_id:2 privileges:<users:<user_proto:"admin" privileges:2 > users:<user_proto:"root" privileges:2 > owner_proto:"root" version:2 > next_mutation_id:1 format_version:3 state:PUBLIC offline_reason:"" view_query:"" is_materialized_view:false new_schema_change_job_id:0 drop_time:0 replacement_of:<id:0 time:<> > audit_mode:DISABLED drop_job_id:0 create_query:"" create_as_of_time:<wall_time:1630565505920835476 > temporary:false partition_all_by:false > due to knob: ignore gossip update
I210902 06:51:45.925654 39392 kv/kvserver/replica_command.go:396  [n1,split,s1,r44/1:/{Table/53-Max}] 1751  initiating a split of this range at key /Table/54 [r45] (span config)
    lease_test.go:1835: expected lease acquisition to not block, but blockCount is: 1
W210902 06:51:45.986681 39617 sql/catalog/lease/lease.go:390  [-] 1752  error: node unavailable; try another peer, not releasing lease: "ID = 54 ver=1 expiration={2021-09-02 06:51:46.008421 +0000 UTC}"
W210902 06:51:45.986915 39243 kv/kvserver/intentresolver/intent_resolver.go:790  [-] 1754  failed to gc transaction record: could not GC completed transaction anchored at /Local/Range/Table/53/RangeDescriptor: node unavailable; try another peer
W210902 06:51:45.986852 39641 sql/catalog/lease/lease.go:390  [-] 1753  error: node unavailable; try another peer, not releasing lease: "ID = 6 ver=1 expiration={2021-09-02 06:51:46.008179 +0000 UTC}"
I210902 06:51:45.986986 38252 sql/catalog/lease/lease.go:1059  [n1] 1755  didnt refresh descriptor: 4 lease: node unavailable; try another peer
I210902 06:51:45.987088 38252 sql/catalog/lease/lease.go:1059  [n1] 1756  didnt refresh descriptor: 35 lease: node unavailable; try another peer
W210902 06:51:45.987173 38253 jobs/registry.go:822  [-] 1757  canceling all adopted jobs due to stopper quiescing
W210902 06:51:45.987154 39645 sql/catalog/lease/storage.go:180  [-] 1758  error releasing lease "ID = 53 ver=1 expiration={2021-09-02 06:51:46.009529 +0000 UTC}": lease-release: failed to send RPC: sending to all replicas failed; last error: node unavailable; try another peer
W210902 06:51:45.987242 39657 sql/catalog/lease/lease.go:390  [-] 1759  error: node unavailable; try another peer, not releasing lease: "ID = 33 ver=1 expiration={2021-09-02 06:51:46.008108 +0000 UTC}"
W210902 06:51:45.987276 39669 sql/catalog/lease/storage.go:180  [-] 1760  error releasing lease "ID = 37 ver=1 expiration={2021-09-02 06:51:46.006183 +0000 UTC}": lease-release: node unavailable; try another peer
W210902 06:51:45.987481 38143 sql/sqlliveness/slinstance/slinstance.go:217  [n1] 1761  exiting heartbeat loop
W210902 06:51:45.987590 39639 sql/catalog/lease/storage.go:180  [-] 1762  error releasing lease "ID = 23 ver=1 expiration={2021-09-02 06:51:46.006139 +0000 UTC}": lease-release: failed to send RPC: sending to all replicas failed; last error: node unavailable; try another peer
E210902 06:51:45.987754 39654 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 1763  making txn commit explicit failed for "unnamed" meta={id=7037980d pri=100.00000000 epo=0 ts=1630565505.982355394,0 min=1630565505.982355394,0 seq=2} lock=true stat=STAGING rts=1630565505.982355394,0 wto=false gul=1630565506.482355394,0 ifw=1: result is ambiguous (server shutdown)
W210902 06:51:45.987965 39671 kv/txn.go:676  [n1] 1764  failure aborting transaction: node unavailable; try another peer; abort caused by: lease-insert: context canceled
E210902 06:51:45.987998 39668 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 1765  making txn commit explicit failed for "unnamed" meta={id=f15e453c pri=100.00000000 epo=0 ts=1630565505.982403894,0 min=1630565505.982403894,0 seq=2} lock=true stat=STAGING rts=1630565505.982403894,0 wto=false gul=1630565506.482403894,0 ifw=1: result is ambiguous (server shutdown)
I210902 06:51:45.988031 39670 sql/catalog/lease/lease.go:1056  [n1] 1766  refreshing descriptor: 15 lease failed: lease-insert: context canceled
W210902 06:51:45.988046 39564 kv/txn.go:676  [intExec=lease-release] 1767  failure aborting transaction: failed to send RPC: sending to all replicas failed; last error: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
E210902 06:51:45.988120 39644 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 1768  making txn commit explicit failed for "unnamed" meta={id=f26684f6 pri=100.00000000 epo=0 ts=1630565505.983927372,0 min=1630565505.983927372,0 seq=2} lock=true stat=STAGING rts=1630565505.983927372,0 wto=false gul=1630565506.483927372,0 ifw=1: result is ambiguous (server shutdown)
W210902 06:51:45.988146 39655 sql/catalog/lease/storage.go:180  [-] 1769  error releasing lease "ID = 21 ver=1 expiration={2021-09-02 06:51:46.006083 +0000 UTC}": lease-release: result is ambiguous (server shutdown)
W210902 06:51:45.988922 39622 kv/txn.go:676  [n1] 1770  failure aborting transaction: node unavailable; try another peer; abort caused by: lease-insert: context canceled
I210902 06:51:45.988986 39609 sql/catalog/lease/lease.go:1056  [n1] 1771  refreshing descriptor: 12 lease failed: lease-insert: context canceled
W210902 06:51:45.992302 39550 kv/txn.go:676  [n1] 1772  failure aborting transaction: node unavailable; try another peer; abort caused by: collecting referenced descriptors: aborted in DistSender: context canceled
I210902 06:51:45.992400 39608 sql/catalog/lease/lease.go:1056  [n1] 1773  refreshing descriptor: 31 lease failed: collecting referenced descriptors: aborted in DistSender: context canceled
--- FAIL: TestLeaseRenewedPeriodically (0.27s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestLeaseRenewedPeriodically PKG=./pkg/sql/catalog/lease TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
vivekmenezes marked as alumn{us,a}; ignoring
```
</p>
</details>

/cc @cockroachdb/sql-schema
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseRenewedPeriodically.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/catalog/lease.TestLeaseRenewedPeriodically [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3465565&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3465565&tab=artifacts#/) on master @ [d49fadb6f1c67d99ce91b719bac44b5640fa8e01](https://github.com/cockroachdb/cockroach/commits/d49fadb6f1c67d99ce91b719bac44b5640fa8e01):


```
I210919 06:43:28.681621 36921 gossip/gossip.go:402  [n1] 1679  NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:38613" > attrs:<> locality:<> ServerVersion:<major_val:21 minor_val:1 patch:0 internal:1164 > build_tag:"v21.2.0-alpha.00000000-4299-gd49fadb6f1" started_at:1632033808681617146 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.0.0.1:44883" >
I210919 06:43:28.687451 36921 server/node.go:424  [n1] 1680  initialized store s1
I210919 06:43:28.687506 36921 kv/kvserver/stores.go:250  [n1] 1681  read 0 node addresses from persistent storage
I210919 06:43:28.687566 36921 server/node.go:501  [n1] 1682  started with engine type 2
I210919 06:43:28.687616 36921 server/node.go:503  [n1] 1683  started with attributes []
I210919 06:43:28.687608 37022 1@server/server.go:1556  [n1] 1684  node connected via gossip
I210919 06:43:28.687737 36921 1@server/server.go:1681  [n1] 1685  starting https server at 127.0.0.1:34565 (use: 127.0.0.1:34565)
I210919 06:43:28.687781 36921 1@server/server.go:1686  [n1] 1686  starting postgres server at 127.0.0.1:44883 (use: 127.0.0.1:44883)
I210919 06:43:28.687814 36921 1@server/server.go:1688  [n1] 1687  starting grpc server at 127.0.0.1:38613
I210919 06:43:28.687844 36921 1@server/server.go:1689  [n1] 1688  advertising CockroachDB node at 127.0.0.1:38613
I210919 06:43:28.692379 36921 1@util/log/event_log.go:32  [n1] 1689 ={"Timestamp":1632033808692377542,"EventType":"node_join","NodeID":1,"StartedAt":1632033808681617146,"LastUp":1632033808681617146}
I210919 06:43:28.692434 36921 sql/sqlliveness/slinstance/slinstance.go:293  [n1] 1690  starting SQL liveness instance
I210919 06:43:28.692775 37161 sql/sqlstats/persistedsqlstats/provider.go:142  [n1] 1691  starting sql-stats-worker with initial delay: 58m15.863042077s
I210919 06:43:28.693091 37156 sql/temporary_schema.go:508  [n1] 1692  running temporary object cleanup background job
I210919 06:43:28.693545 37156 sql/temporary_schema.go:552  [n1] 1693  found 0 temporary schemas
I210919 06:43:28.693579 37156 sql/temporary_schema.go:555  [n1] 1694  early exiting temporary schema cleaner as no temporary schemas were found
I210919 06:43:28.693600 37156 sql/temporary_schema.go:556  [n1] 1695  completed temporary object cleanup job
I210919 06:43:28.693609 37154 sql/sqlliveness/slstorage/slstorage.go:393  [n1] 1696  inserted sqlliveness session be63c8923f0f451e934facb9e37400be
I210919 06:43:28.693662 37154 sql/sqlliveness/slinstance/slinstance.go:180  [n1] 1698  created new SQL liveness session be63c8923f0f451e934facb9e37400be
I210919 06:43:28.693623 37156 sql/temporary_schema.go:639  [n1] 1697  temporary object cleaner next scheduled to run at 2021-09-19 07:13:28.693077812 +0000 UTC
I210919 06:43:28.702262 37317 kv/kvserver/replica_rangefeed.go:649  [n1,rangefeed=lease,s1,r6/1:/Table/{SystemCon…-11}] 1699  RangeFeed closed timestamp is empty
I210919 06:43:28.707365 37277 util/log/event_log.go:32  [n1,intExec=optInToDiagnosticsStatReporting] 1700 ={"Timestamp":1632033808703007737,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"diagnostics.reporting.enabled\" = true","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-optInToDiagnosticsStatReporting","SettingName":"diagnostics.reporting.enabled","Value":"true"}
I210919 06:43:28.710331 37380 migration/migrationmanager/manager.go:110  [n1,intExec=set-setting,migration-mgr] 1701  no need to migrate, cluster already at newest version
I210919 06:43:28.711363 37380 util/log/event_log.go:32  [n1,intExec=set-setting] 1702 ={"Timestamp":1632033808709263791,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-set-setting","PlaceholderValues":["'21.1-1164'"],"SettingName":"version","Value":"21.1-1164"}
W210919 06:43:28.718527 36952 kv/range_lookup.go:243  [n1,rangefeed=table-stats-cache,range-lookup=/Meta2/Table/20/NULL] 1703  range lookup of key /Meta2/Table/20/NULL found only non-matching ranges []; retrying
W210919 06:43:28.719902 36951 kv/range_lookup.go:243  [n1,rangefeed=table-stats-cache,range-lookup=/Table/20] 1704  range lookup of key /Table/20 found only non-matching ranges []; retrying
I210919 06:43:28.722305 37548 util/log/event_log.go:32  [n1,intExec=initializeClusterSecret] 1705 ={"Timestamp":1632033808719622076,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"cluster.secret\" = gen_random_uuid()::STRING","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-initializeClusterSecret","SettingName":"cluster.secret","Value":"646c5b71-5a8c-46a0-8064-4a0919c0ad96"}
I210919 06:43:28.729414 37552 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 1706 ={"Timestamp":1632033808725074263,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS defaultdb","Tag":"CREATE DATABASE","User":"root","DescriptorID":50,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"defaultdb"}
I210919 06:43:28.729855 37301 sql/catalog/lease/lease.go:906  [n1] 1707  skipping update of database:<name:"defaultdb" id:50 modification_time:<wall_time:1632033808725074263 > version:1 privileges:<users:<user_proto:"admin" privileges:2 > users:<user_proto:"root" privileges:2 > owner_proto:"root" version:2 > state:PUBLIC offline_reason:"" default_privileges:<> > due to knob: ignore gossip update
I210919 06:43:28.732355 37301 sql/catalog/lease/lease.go:906  [n1] 1708  skipping update of database:<name:"postgres" id:51 modification_time:<wall_time:1632033808729688097 > version:1 privileges:<users:<user_proto:"admin" privileges:2 > users:<user_proto:"root" privileges:2 > owner_proto:"root" version:2 > state:PUBLIC offline_reason:"" default_privileges:<> > due to knob: ignore gossip update
I210919 06:43:28.732576 37844 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 1709 ={"Timestamp":1632033808729688097,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS postgres","Tag":"CREATE DATABASE","User":"root","DescriptorID":51,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"postgres"}
I210919 06:43:28.732726 37021 3@vendor/github.com/cockroachdb/pebble/event.go:631  [n1,pebble,s1] 1710  [JOB 3] WAL created 000004
I210919 06:43:28.754820 36921 server/server_sql.go:1093  [n1] 1711  done ensuring all necessary startup migrations have run
I210919 06:43:28.755658 37697 jobs/job_scheduler.go:357  [n1] 1712  waiting 2m0s before scheduled jobs daemon start
I210919 06:43:28.758031 38131 server/auto_upgrade.go:55  [n1] 1713  no need to upgrade, cluster already at the newest version
I210919 06:43:28.767747 36921 1@server/server.go:2338  [n1] 1714  serving sql connections
I210919 06:43:28.778255 37301 sql/catalog/lease/lease.go:906  [n1] 1715  skipping update of database:<name:"t" id:52 modification_time:<wall_time:1632033808775045825 > version:1 privileges:<users:<user_proto:"admin" privileges:2 > users:<user_proto:"root" privileges:2 > owner_proto:"root" version:2 > state:PUBLIC offline_reason:"" default_privileges:<> > due to knob: ignore gossip update
I210919 06:43:28.778562 38129 5@util/log/event_log.go:32  [n1,client=127.0.0.1:37166,hostssl,user=root] 1716 ={"Timestamp":1632033808775045825,"EventType":"create_database","Statement":"CREATE DATABASE t","Tag":"CREATE DATABASE","User":"root","DescriptorID":52,"DatabaseName":"t"}
I210919 06:43:28.787973 38129 5@util/log/event_log.go:32  [n1,client=127.0.0.1:37166,hostssl,user=root] 1717 ={"Timestamp":1632033808778873714,"EventType":"create_table","Statement":"CREATE TABLE t.public.test1 (k CHAR PRIMARY KEY, v CHAR)","Tag":"CREATE TABLE","User":"root","DescriptorID":53,"TableName":"t.public.test1"}
I210919 06:43:28.787959 37301 sql/catalog/lease/lease.go:906  [n1] 1718  skipping update of table:<name:"test1" id:53 version:1 modification_time:<wall_time:1632033808778873714 > parent_id:52 unexposed_parent_schema_id:29 columns:<name:"k" id:1 type:<InternalType:<family:StringFamily width:1 precision:0 locale:"" visible_type:0 oid:1042 time_precision_is_set:false > TypeMeta:<Version:0 > > nullable:false hidden:false inaccessible:false generated_as_identity_type:NOT_IDENTITY_COLUMN virtual:false pg_attribute_num:0 alter_column_type_in_progress:false system_column_kind:NONE > columns:<name:"v" id:2 type:<InternalType:<family:StringFamily width:1 precision:0 locale:"" visible_type:0 oid:1042 time_precision_is_set:false > TypeMeta:<Version:0 > > nullable:true hidden:false inaccessible:false generated_as_identity_type:NOT_IDENTITY_COLUMN virtual:false pg_attribute_num:0 alter_column_type_in_progress:false system_column_kind:NONE > next_column_id:3 families:<name:"primary" id:0 column_names:"k" column_names:"v" column_ids:1 column_ids:2 default_column_id:2 > next_family_id:1 primary_index:<name:"primary" id:1 unique:true version:4 key_column_names:"k" key_column_directions:ASC store_column_names:"v" key_column_ids:1 store_column_ids:2 foreign_key:<table:0 index:0 name:"" validity:Validated shared_prefix_len:0 on_delete:NO_ACTION on_update:NO_ACTION match:SIMPLE > interleave:<> partitioning:<num_columns:0 num_implicit_columns:0 > type:FORWARD created_explicitly:false encoding_type:1 sharded:<is_sharded:false name:"" shard_buckets:0 > disabled:false geo_config:<> predicate:"" > next_index_id:2 privileges:<users:<user_proto:"admin" privileges:2 > users:<user_proto:"root" privileges:2 > owner_proto:"root" version:2 > next_mutation_id:1 format_version:3 state:PUBLIC offline_reason:"" view_query:"" is_materialized_view:false new_schema_change_job_id:0 drop_time:0 replacement_of:<id:0 time:<> > audit_mode:DISABLED drop_job_id:0 create_query:"" create_as_of_time:<wall_time:1632033808778873714 > temporary:false partition_all_by:false > due to knob: ignore gossip update
I210919 06:43:28.789596 38152 kv/kvserver/replica_command.go:395  [n1,split,s1,r43/1:/{Table/47-Max}] 1719  initiating a split of this range at key /Table/53 [r44] (span config)
I210919 06:43:28.794500 38129 5@util/log/event_log.go:32  [n1,client=127.0.0.1:37166,hostssl,user=root] 1720 ={"Timestamp":1632033808788153760,"EventType":"create_table","Statement":"CREATE TABLE t.public.test2 ()","Tag":"CREATE TABLE","User":"root","DescriptorID":54,"TableName":"t.public.test2"}
I210919 06:43:28.794994 37301 sql/catalog/lease/lease.go:906  [n1] 1721  skipping update of table:<name:"test2" id:54 version:1 modification_time:<wall_time:1632033808788153760 > parent_id:52 unexposed_parent_schema_id:29 columns:<name:"rowid" id:1 type:<InternalType:<family:IntFamily width:64 precision:0 locale:"" visible_type:0 oid:20 time_precision_is_set:false > TypeMeta:<Version:0 > > nullable:false default_expr:"unique_rowid()" hidden:true inaccessible:false generated_as_identity_type:NOT_IDENTITY_COLUMN virtual:false pg_attribute_num:0 alter_column_type_in_progress:false system_column_kind:NONE > next_column_id:2 families:<name:"primary" id:0 column_names:"rowid" column_ids:1 default_column_id:0 > next_family_id:1 primary_index:<name:"primary" id:1 unique:true version:4 key_column_names:"rowid" key_column_directions:ASC key_column_ids:1 foreign_key:<table:0 index:0 name:"" validity:Validated shared_prefix_len:0 on_delete:NO_ACTION on_update:NO_ACTION match:SIMPLE > interleave:<> partitioning:<num_columns:0 num_implicit_columns:0 > type:FORWARD created_explicitly:false encoding_type:1 sharded:<is_sharded:false name:"" shard_buckets:0 > disabled:false geo_config:<> predicate:"" > next_index_id:2 privileges:<users:<user_proto:"admin" privileges:2 > users:<user_proto:"root" privileges:2 > owner_proto:"root" version:2 > next_mutation_id:1 format_version:3 state:PUBLIC offline_reason:"" view_query:"" is_materialized_view:false new_schema_change_job_id:0 drop_time:0 replacement_of:<id:0 time:<> > audit_mode:DISABLED drop_job_id:0 create_query:"" create_as_of_time:<wall_time:1632033808788153760 > temporary:false partition_all_by:false > due to knob: ignore gossip update
I210919 06:43:28.797107 38359 kv/kvserver/replica_command.go:395  [n1,split,s1,r44/1:/{Table/53-Max}] 1722  initiating a split of this range at key /Table/54 [r45] (span config)
I210919 06:43:28.863818 37064 3@vendor/github.com/cockroachdb/pebble/event.go:631  [n1,pebble,s1] 1723  [JOB 4] WAL created 000005
    lease_test.go:1835: expected lease acquisition to not block, but blockCount is: 1
W210919 06:43:28.891700 37693 kv/kvserver/intentresolver/intent_resolver.go:790  [-] 1724  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
W210919 06:43:28.891814 37303 jobs/registry.go:822  [-] 1725  canceling all adopted jobs due to stopper quiescing
W210919 06:43:28.891841 37154 sql/sqlliveness/slinstance/slinstance.go:217  [n1] 1726  exiting heartbeat loop
--- FAIL: TestLeaseRenewedPeriodically (0.27s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestLeaseRenewedPeriodically PKG=./pkg/sql/catalog/lease TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
vivekmenezes marked as alumn{us,a}; ignoring
```
</p>
</details>

/cc @cockroachdb/sql-schema
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseRenewedPeriodically.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/catalog/lease.TestLeaseRenewedPeriodically [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3480641&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3480641&tab=artifacts#/) on master @ [1dbea6a3e4c93ba7c4a43ad34369c438558eba8a](https://github.com/cockroachdb/cockroach/commits/1dbea6a3e4c93ba7c4a43ad34369c438558eba8a):


```
I210922 06:49:29.913109 37058 kv/kvserver/stores.go:250  [n1] 1709  read 0 node addresses from persistent storage
I210922 06:49:29.913155 37058 server/node.go:501  [n1] 1710  started with engine type 2
I210922 06:49:29.913183 37058 server/node.go:503  [n1] 1711  started with attributes []
I210922 06:49:29.914157 37058 1@server/server.go:1681  [n1] 1713  starting https server at 127.0.0.1:32819 (use: 127.0.0.1:32819)
I210922 06:49:29.914224 37058 1@server/server.go:1686  [n1] 1714  starting postgres server at 127.0.0.1:43133 (use: 127.0.0.1:43133)
I210922 06:49:29.914272 37058 1@server/server.go:1688  [n1] 1715  starting grpc server at 127.0.0.1:32913
I210922 06:49:29.914303 37058 1@server/server.go:1689  [n1] 1716  advertising CockroachDB node at 127.0.0.1:32913
I210922 06:49:29.913331 37126 1@server/server.go:1556  [n1] 1712  node connected via gossip
W210922 06:49:29.951680 37031 kv/range_lookup.go:243  [n1,rangefeed=table-stats-cache,range-lookup=/Meta2/Table/20/NULL] 1717  range lookup of key /Meta2/Table/20/NULL found only non-matching ranges []; retrying
W210922 06:49:29.953111 37030 kv/range_lookup.go:243  [n1,rangefeed=table-stats-cache,range-lookup=/Table/20] 1718  range lookup of key /Table/20 found only non-matching ranges []; retrying
I210922 06:49:30.006223 37072 kv/kvserver/replica_rangefeed.go:649  [n1,rangefeed=table-stats-cache,s1,r16/1:/Table/2{0-1}] 1719  RangeFeed closed timestamp is empty
I210922 06:49:30.070981 37058 1@util/log/event_log.go:32  [n1] 1720 ={"Timestamp":1632293370070979098,"EventType":"node_join","NodeID":1,"StartedAt":1632293369910790469,"LastUp":1632293369910790469}
I210922 06:49:30.071037 37058 sql/sqlliveness/slinstance/slinstance.go:293  [n1] 1721  starting SQL liveness instance
I210922 06:49:30.071123 37349 sql/temporary_schema.go:508  [n1] 1722  running temporary object cleanup background job
I210922 06:49:30.071189 37354 sql/sqlstats/persistedsqlstats/provider.go:142  [n1] 1723  starting sql-stats-worker with initial delay: 1h2m6.057272393s
I210922 06:49:30.072185 37347 sql/sqlliveness/slstorage/slstorage.go:393  [n1] 1724  inserted sqlliveness session dfaec0ea88de41a983da6e6d47c30abc
I210922 06:49:30.072232 37347 sql/sqlliveness/slinstance/slinstance.go:180  [n1] 1725  created new SQL liveness session dfaec0ea88de41a983da6e6d47c30abc
I210922 06:49:30.072357 37349 sql/temporary_schema.go:552  [n1] 1726  found 0 temporary schemas
I210922 06:49:30.072396 37349 sql/temporary_schema.go:555  [n1] 1727  early exiting temporary schema cleaner as no temporary schemas were found
I210922 06:49:30.072417 37349 sql/temporary_schema.go:556  [n1] 1728  completed temporary object cleanup job
I210922 06:49:30.072439 37349 sql/temporary_schema.go:639  [n1] 1729  temporary object cleaner next scheduled to run at 2021-09-22 07:19:30.071108036 +0000 UTC
I210922 06:49:30.082746 37388 util/log/event_log.go:32  [n1,intExec=optInToDiagnosticsStatReporting] 1730 ={"Timestamp":1632293370078676789,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"diagnostics.reporting.enabled\" = true","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-optInToDiagnosticsStatReporting","SettingName":"diagnostics.reporting.enabled","Value":"true"}
I210922 06:49:30.085798 37553 migration/migrationmanager/manager.go:110  [n1,intExec=set-setting,migration-mgr] 1731  no need to migrate, cluster already at newest version
I210922 06:49:30.086828 37553 util/log/event_log.go:32  [n1,intExec=set-setting] 1732 ={"Timestamp":1632293370084716630,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-set-setting","PlaceholderValues":["'21.1-1164'"],"SettingName":"version","Value":"21.1-1164"}
I210922 06:49:30.096879 37860 util/log/event_log.go:32  [n1,intExec=initializeClusterSecret] 1733 ={"Timestamp":1632293370094388079,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"cluster.secret\" = gen_random_uuid()::STRING","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-initializeClusterSecret","SettingName":"cluster.secret","Value":"64cc4b29-4430-4ce5-a6e1-132b7a0dfcaf"}
I210922 06:49:30.099958 37605 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 1734 ={"Timestamp":1632293370098047928,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS defaultdb","Tag":"CREATE DATABASE","User":"root","DescriptorID":50,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"defaultdb"}
I210922 06:49:30.101687 37433 sql/catalog/lease/lease.go:906  [n1] 1735  skipping update of database:<name:"defaultdb" id:50 modification_time:<wall_time:1632293370098047928 > version:1 privileges:<users:<user_proto:"admin" privileges:2 > users:<user_proto:"root" privileges:2 > owner_proto:"root" version:2 > state:PUBLIC offline_reason:"" default_privileges:<> > due to knob: ignore gossip update
I210922 06:49:30.102481 37611 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 1736 ={"Timestamp":1632293370100167951,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS postgres","Tag":"CREATE DATABASE","User":"root","DescriptorID":51,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"postgres"}
I210922 06:49:30.102509 37433 sql/catalog/lease/lease.go:906  [n1] 1737  skipping update of database:<name:"postgres" id:51 modification_time:<wall_time:1632293370100167951 > version:1 privileges:<users:<user_proto:"admin" privileges:2 > users:<user_proto:"root" privileges:2 > owner_proto:"root" version:2 > state:PUBLIC offline_reason:"" default_privileges:<> > due to knob: ignore gossip update
I210922 06:49:30.105080 37158 3@vendor/github.com/cockroachdb/pebble/event.go:631  [n1,pebble,s1] 1738  [JOB 3] WAL created 000004
I210922 06:49:30.117451 37058 server/server_sql.go:1093  [n1] 1739  done ensuring all necessary startup migrations have run
I210922 06:49:30.117542 38062 jobs/job_scheduler.go:357  [n1] 1740  waiting 4m0s before scheduled jobs daemon start
I210922 06:49:30.119341 38064 server/auto_upgrade.go:55  [n1] 1741  no need to upgrade, cluster already at the newest version
I210922 06:49:30.131913 37058 1@server/server.go:2338  [n1] 1742  serving sql connections
I210922 06:49:30.141557 37433 sql/catalog/lease/lease.go:906  [n1] 1743  skipping update of database:<name:"t" id:52 modification_time:<wall_time:1632293370139060063 > version:1 privileges:<users:<user_proto:"admin" privileges:2 > users:<user_proto:"root" privileges:2 > owner_proto:"root" version:2 > state:PUBLIC offline_reason:"" default_privileges:<> > due to knob: ignore gossip update
I210922 06:49:30.141636 38234 5@util/log/event_log.go:32  [n1,client=127.0.0.1:46884,hostssl,user=root] 1744 ={"Timestamp":1632293370139060063,"EventType":"create_database","Statement":"CREATE DATABASE t","Tag":"CREATE DATABASE","User":"root","DescriptorID":52,"DatabaseName":"t"}
I210922 06:49:30.149726 38234 5@util/log/event_log.go:32  [n1,client=127.0.0.1:46884,hostssl,user=root] 1745 ={"Timestamp":1632293370141780445,"EventType":"create_table","Statement":"CREATE TABLE t.public.test1 (k CHAR PRIMARY KEY, v CHAR)","Tag":"CREATE TABLE","User":"root","DescriptorID":53,"TableName":"t.public.test1"}
I210922 06:49:30.149700 37433 sql/catalog/lease/lease.go:906  [n1] 1746  skipping update of table:<name:"test1" id:53 version:1 modification_time:<wall_time:1632293370141780445 > parent_id:52 unexposed_parent_schema_id:29 columns:<name:"k" id:1 type:<InternalType:<family:StringFamily width:1 precision:0 locale:"" visible_type:0 oid:1042 time_precision_is_set:false > TypeMeta:<Version:0 > > nullable:false hidden:false inaccessible:false generated_as_identity_type:NOT_IDENTITY_COLUMN virtual:false pg_attribute_num:0 alter_column_type_in_progress:false system_column_kind:NONE > columns:<name:"v" id:2 type:<InternalType:<family:StringFamily width:1 precision:0 locale:"" visible_type:0 oid:1042 time_precision_is_set:false > TypeMeta:<Version:0 > > nullable:true hidden:false inaccessible:false generated_as_identity_type:NOT_IDENTITY_COLUMN virtual:false pg_attribute_num:0 alter_column_type_in_progress:false system_column_kind:NONE > next_column_id:3 families:<name:"primary" id:0 column_names:"k" column_names:"v" column_ids:1 column_ids:2 default_column_id:2 > next_family_id:1 primary_index:<name:"primary" id:1 unique:true version:4 key_column_names:"k" key_column_directions:ASC store_column_names:"v" key_column_ids:1 store_column_ids:2 foreign_key:<table:0 index:0 name:"" validity:Validated shared_prefix_len:0 on_delete:NO_ACTION on_update:NO_ACTION match:SIMPLE > interleave:<> partitioning:<num_columns:0 num_implicit_columns:0 > type:FORWARD created_explicitly:false encoding_type:1 sharded:<is_sharded:false name:"" shard_buckets:0 > disabled:false geo_config:<> predicate:"" > next_index_id:2 privileges:<users:<user_proto:"admin" privileges:2 > users:<user_proto:"root" privileges:2 > owner_proto:"root" version:2 > next_mutation_id:1 format_version:3 state:PUBLIC offline_reason:"" view_query:"" is_materialized_view:false new_schema_change_job_id:0 drop_time:0 replacement_of:<id:0 time:<> > audit_mode:DISABLED drop_job_id:0 create_query:"" create_as_of_time:<wall_time:1632293370141780445 > temporary:false partition_all_by:false > due to knob: ignore gossip update
I210922 06:49:30.151361 38390 kv/kvserver/replica_command.go:395  [n1,split,s1,r43/1:/{Table/47-Max}] 1747  initiating a split of this range at key /Table/53 [r44] (span config)
I210922 06:49:30.153628 38234 5@util/log/event_log.go:32  [n1,client=127.0.0.1:46884,hostssl,user=root] 1748 ={"Timestamp":1632293370149944424,"EventType":"create_table","Statement":"CREATE TABLE t.public.test2 ()","Tag":"CREATE TABLE","User":"root","DescriptorID":54,"TableName":"t.public.test2"}
I210922 06:49:30.153998 37433 sql/catalog/lease/lease.go:906  [n1] 1749  skipping update of table:<name:"test2" id:54 version:1 modification_time:<wall_time:1632293370149944424 > parent_id:52 unexposed_parent_schema_id:29 columns:<name:"rowid" id:1 type:<InternalType:<family:IntFamily width:64 precision:0 locale:"" visible_type:0 oid:20 time_precision_is_set:false > TypeMeta:<Version:0 > > nullable:false default_expr:"unique_rowid()" hidden:true inaccessible:false generated_as_identity_type:NOT_IDENTITY_COLUMN virtual:false pg_attribute_num:0 alter_column_type_in_progress:false system_column_kind:NONE > next_column_id:2 families:<name:"primary" id:0 column_names:"rowid" column_ids:1 default_column_id:0 > next_family_id:1 primary_index:<name:"primary" id:1 unique:true version:4 key_column_names:"rowid" key_column_directions:ASC key_column_ids:1 foreign_key:<table:0 index:0 name:"" validity:Validated shared_prefix_len:0 on_delete:NO_ACTION on_update:NO_ACTION match:SIMPLE > interleave:<> partitioning:<num_columns:0 num_implicit_columns:0 > type:FORWARD created_explicitly:false encoding_type:1 sharded:<is_sharded:false name:"" shard_buckets:0 > disabled:false geo_config:<> predicate:"" > next_index_id:2 privileges:<users:<user_proto:"admin" privileges:2 > users:<user_proto:"root" privileges:2 > owner_proto:"root" version:2 > next_mutation_id:1 format_version:3 state:PUBLIC offline_reason:"" view_query:"" is_materialized_view:false new_schema_change_job_id:0 drop_time:0 replacement_of:<id:0 time:<> > audit_mode:DISABLED drop_job_id:0 create_query:"" create_as_of_time:<wall_time:1632293370149944424 > temporary:false partition_all_by:false > due to knob: ignore gossip update
I210922 06:49:30.155029 38562 kv/kvserver/replica_command.go:395  [n1,split,s1,r44/1:/{Table/53-Max}] 1750  initiating a split of this range at key /Table/54 [r45] (span config)
    lease_test.go:1800: expected 1 lease to be acquired, but acquired 2 times
W210922 06:49:30.171236 37347 sql/sqlliveness/slinstance/slinstance.go:217  [n1] 1751  exiting heartbeat loop
W210922 06:49:30.171958 38642 kv/kvserver/intentresolver/intent_resolver.go:790  [-] 1752  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
E210922 06:49:30.171306 38465 kv/kvclient/kvcoord/txn_interceptor_committer.go:479  [-] 1753  making txn commit explicit failed for "unnamed" meta={id=d6f86f4f key=/Table/11/1/53/1/2021-09-22T06:49:30.214782Z/1/0 pri=100.00000000 epo=0 ts=1632293370.164781726,0 min=1632293370.164781726,0 seq=2} lock=true stat=STAGING rts=1632293370.164781726,0 wto=false gul=1632293370.664781726,0 ifw=1: failed to send RPC: sending to all replicas failed; last error: node unavailable; try another peer
W210922 06:49:30.172102 38675 kv/txn.go:687  [intExec=lease-release] 1754  failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
W210922 06:49:30.172159 37435 jobs/registry.go:822  [-] 1755  canceling all adopted jobs due to stopper quiescing
W210922 06:49:30.172228 38659 sql/catalog/lease/storage.go:180  [-] 1756  error releasing lease "ID = 4 ver=1 expiration={2021-09-22 06:49:30.178118 +0000 UTC}": lease-release: result is ambiguous (server shutdown)
--- FAIL: TestLeaseRenewedPeriodically (0.30s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestLeaseRenewedPeriodically PKG=./pkg/sql/catalog/lease TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
vivekmenezes marked as alumn{us,a}; ignoring
```
</p>
</details>

/cc @cockroachdb/sql-schema
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseRenewedPeriodically.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/catalog/lease/lease.go
+++ b/pkg/sql/catalog/lease/lease.go
@@ -299,7 +299,7 @@ func (m *Manager) AcquireFreshestFromStore(ctx context.Context, id descpb.ID) er
 		// Acquire a fresh lease.
 		didAcquire, err := acquireNodeLease(ctx, m, id)
 		if m.testingKnobs.LeaseStoreTestingKnobs.LeaseAcquireResultBlockEvent != nil {
-			m.testingKnobs.LeaseStoreTestingKnobs.LeaseAcquireResultBlockEvent(AcquireFreshestBlock)
+			m.testingKnobs.LeaseStoreTestingKnobs.LeaseAcquireResultBlockEvent(AcquireFreshestBlock, id)
 		}
 		if err != nil {
 			return err
@@ -814,7 +814,7 @@ func (m *Manager) Acquire(
 			}
 
 			if m.testingKnobs.LeaseStoreTestingKnobs.LeaseAcquireResultBlockEvent != nil {
-				m.testingKnobs.LeaseStoreTestingKnobs.LeaseAcquireResultBlockEvent(AcquireBlock)
+				m.testingKnobs.LeaseStoreTestingKnobs.LeaseAcquireResultBlockEvent(AcquireBlock, id)
 			}
 
 		case errors.Is(err, errReadOlderVersion):

--- a/pkg/sql/catalog/lease/lease_internal_test.go
+++ b/pkg/sql/catalog/lease/lease_internal_test.go
@@ -945,7 +945,7 @@ func TestLeaseAcquireAndReleaseConcurrently(t *testing.T) {
 					LeaseStoreTestingKnobs: StorageTestingKnobs{
 						RemoveOnceDereferenced: true,
 						LeaseReleasedEvent:     removalTracker.LeaseRemovedNotification,
-						LeaseAcquireResultBlockEvent: func(leaseBlockType AcquireBlockType) {
+						LeaseAcquireResultBlockEvent: func(leaseBlockType AcquireBlockType, _ descpb.ID) {
 							if leaseBlockType == AcquireBlock {
 								if count := atomic.LoadInt32(&acquireArrivals); (count < 1 && test.isSecondCallAcquireFreshest) ||
 									(count < 2 && !test.isSecondCallAcquireFreshest) {

--- a/pkg/sql/catalog/lease/lease_test.go
+++ b/pkg/sql/catalog/lease/lease_test.go
@@ -1282,8 +1282,10 @@ func TestLeaseRenewedAutomatically(testingT *testing.T) {
 						atomic.AddInt32(&testAcquiredCount, 1)
 					}
 				},
-				LeaseAcquireResultBlockEvent: func(_ lease.AcquireBlockType) {
-					atomic.AddInt32(&testAcquisitionBlockCount, 1)
+				LeaseAcquireResultBlockEvent: func(_ lease.AcquireBlockType, id descpb.ID) {
+					if id > keys.MaxReservedDescID {
+						atomic.AddInt32(&testAcquisitionBlockCount, 1)
+					}
 				},
 			},
 		},
@@ -1739,8 +1741,10 @@ func TestLeaseRenewedPeriodically(testingT *testing.T) {
 					defer mu.Unlock()
 					releasedIDs[id] = struct{}{}
 				},
-				LeaseAcquireResultBlockEvent: func(_ lease.AcquireBlockType) {
-					atomic.AddInt32(&testAcquisitionBlockCount, 1)
+				LeaseAcquireResultBlockEvent: func(_ lease.AcquireBlockType, id descpb.ID) {
+					if id > keys.MaxReservedDescID {
+						atomic.AddInt32(&testAcquisitionBlockCount, 1)
+					}
 				},
 			},
 			TestingDescriptorUpdateEvent: func(_ *descpb.Descriptor) error {

--- a/pkg/sql/catalog/lease/testutils.go
+++ b/pkg/sql/catalog/lease/testutils.go
@@ -30,7 +30,7 @@ type StorageTestingKnobs struct {
 	LeaseAcquiredEvent func(desc catalog.Descriptor, err error)
 	// Called before waiting on a results from a DoChan call of acquireNodeLease
 	// in descriptorState.acquire() and descriptorState.acquireFreshestFromStore().
-	LeaseAcquireResultBlockEvent func(leaseBlockType AcquireBlockType)
+	LeaseAcquireResultBlockEvent func(leaseBlockType AcquireBlockType, id descpb.ID)
 	// RemoveOnceDereferenced forces leases to be removed
 	// as soon as they are dereferenced.
 	RemoveOnceDereferenced bool

